be/src/exec/hdfs_scanner_orc.cpp

```diff
@@ -548,6 +554,30 @@ Status HdfsOrcScanner::do_get_next(RuntimeState* runtime_state, ChunkPtr* chunk)
     return Status::OK();
 }
 
+StatusOr<size_t> HdfsOrcScanner::_do_get_next_count(ChunkPtr* chunk) {
+    size_t read_num_values = 0;
+    Status st = Status::OK();
+    while (true) {
+        {
+            SCOPED_RAW_TIMER(&_app_stats.column_read_ns);
+            orc::RowReader::ReadPosition position;
+            st = _orc_reader->read_next(&position);
+            if (!st.ok()) {
+                break;
+            }
+        }
+        read_num_values += _orc_reader->get_cvb_size();
+        if (!_need_skip_rowids.empty()) {
+            read_num_values -= _orc_reader->get_row_delete_number(_need_skip_rowids);
+        }
+    }
+
+    if (!st.is_end_of_file()) return st;
+    if (read_num_values == 0) return Status::EndOfFile("No more rows to read");
```

Comment:
`read_num_values == 0` is meaning for EOF?

`_need_skip_rowids` may just filter all records in cvb, but it is not EOF.

Reply:
see the above `while(true)` loop
- it will constantly read till eof, and break
- so right here, we have two cases
   -  st = eof
   -  st != eof(return)
- and if `(read_num_values == 0)`,   then we can return eof
- but if not, it means we have a chunk to process.